expo/snack
# [snackager] Upgrade to webpack@5 and latest babel/metro

# Why

Upgrade to webpack@5, newer babel dependencies and the latest `metro-react-native-babel-preset`. To better support newer compilation scenarios; and this also produces smaller bundles.

# How

- Update dependencies to `webpack@5`
- Update to latest `metro-react-native-babel-preset`
- Update to latest `@babel/core` & `@babel/generator`
- Upgrade code where needed
- Remove no longer needed webpack@4 dependencies
- Implement babel rewrite plugin for fixing `@material-ui/core` `@babel/runtime/helper/esm/...` imports
- Update test snapshots
- Investigated whether `AssetResolver` was still needed and added comments

# Test Plan

- All tests pass
- Uses cache-busting to regenerate the following bundles and verify them using a Snack example:
  - [x] Verified `react-native-webview`
  - [x] Verified `@react-native-community/datetimepicker`
  - [x] Verified `@react-navigation/native`
  - [x] Verified `@react-navigation/stack`
  - [x] Verified `expo-linear-gradient`
  - [x] Verified `@draftbit/ui`
